src/selectedtests/datasource/datasource_cli.py

```diff
@@ -30,35 +30,49 @@ def setup_mappings_indexes(collection: Collection) -> None:
 
     :param collection: Collection to add indexes to.
     """
-    index = IndexModel([("source_file", ASCENDING)], unique=True)
+    # project, source_file on it's own could be unique, but the repo and branch are needed when
+    # there is a module.
+    index = IndexModel(
+        [
+            ("project", ASCENDING),
+            ("repo", ASCENDING),
+            ("branch", ASCENDING),
+            ("source_file", ASCENDING),
+        ],
+        unique=True,
+    )
     collection.create_indexes([index])
     LOGGER.info("Adding indexes for collection", collection=collection.name)
 
 
-def setup_mappings_test_files_indexes(collection: Collection) -> None:
+def setup_mappings_tasks_indexes(collection: Collection) -> None:
     """
-    Create appropriate indexes for the mapping test files collection.
+    Create appropriate indexes for the mapping tasks collection.
 
     The indexes must support both the $lookup operation and uniqueness constraints.
 
     :param collection: Collection to add indexes to.
     """
-    index = IndexModel([("source_file", ASCENDING), ("name", ASCENDING)], unique=True)
+    # task_mapping_id simplifies the index (but then requires a find_and_update_one for the
```

Comment:
nit: `find_one_and_update` (I got confused when I grepped for `find_and_update_one`)